foqus_lib/gui/tests/test_ml_ai.py

```diff
@@ -0,0 +1,149 @@
+    yield models_dir
+
+
+@pytest.fixture(scope="module")
+def model_files(
+    models_dir: Path,
+    suffixes: Tuple[str] = (".py", ".h5"),
+) -> List[Path]:
+    paths = []
+    for path in sorted(models_dir.glob("*")):
+        if all(
+            [
+                path.is_file(),
+                path.stat().st_size > 0,
+                path.suffix in suffixes,
+                path.name != "__init__.py",
+            ]
+        ):
+            paths.append(path)
+    return paths
+
+
+def test_model_files_are_present(model_files: List[Path]):
+    assert model_files
+
+
+class TestMLAIPluginFlowsheetRun:
+    @pytest.fixture
+    def focus_flowsheet_tab(
+        self,
+        qtbot: QtBot,
+        main_window,
+    ) -> None:
+        qtbot.focused = main_window
+        with qtbot.wait_signal(main_window.fsEditAction.triggered):
+            qtbot.click(button="Flowsheet")
+
+    def test_flowsheet_tab_is_active(
+        self, qtbot: QtBot, main_window, focus_flowsheet_tab: drawFlowsheet
+    ):
+        assert main_window.mainWidget.currentIndex() == main_window.screenIndex["flow"]
+
+    @pytest.fixture
+    def active_session(
+        self,
+        main_window: mainWindow,
+        flowsheet_session_file: Path,
+    ) -> FoqusSession:
+        main_window.loadSessionFile(str(flowsheet_session_file), saveCurrent=False)
+        return main_window.dat
+
+    def test_model_flowsheet_is_loaded(
+        self, qtbot: QtBot, focus_flowsheet_tab, active_session: FoqusSession
+    ):
+        assert active_session.flowsheet is not None
+
+    @pytest.fixture
+    def pymodels_ml_ai(
+        self,
+        active_session: FoqusSession,
+        model_files: List[Path],
+        models_dir: Path,
+    ) -> ml_ai_models:
+        if not model_files:
+            pytest.skip(f"No model files found in directory: {models_dir}")
+        return active_session.pymodels_ml_ai
+
+    def test_ml_ai_models_loaded(self, pymodels_ml_ai: ml_ai_models):
+        assert len(pymodels_ml_ai.ml_ai_models) > 0
+
+    @pytest.fixture
+    def trigger_flowsheet_run_action(
+        self,
+        qtbot: QtBot,
+        active_session,
+        main_window: mainWindow,
+        pymodels_ml_ai,
+    ):
+        run_action = main_window.runAction
+        with qtbot.replacing_with_signal(
+            (QtWidgets.QMessageBox, "information"),
+            (QtWidgets.QMessageBox, "critical"),
+        ) as signal:
+            with qtbot.wait_signal(signal, timeout=2_000):
+                run_action.trigger()
+        return run_action
+
+    @pytest.fixture(scope="function")
+    def statusbar_message(self, main_window: mainWindow) -> str:
+        return main_window.statusBar().currentMessage()
+
+    def test_flowsheet_run_successful(
+        self,
+        trigger_flowsheet_run_action,
+        statusbar_message: str,
+        text_when_success: str = "Finished Single Simulation... Success",
+    ):
+        assert text_when_success in statusbar_message
```

Comment:
This all looks very nice, excellent job!

Reply:
Thanks! Once we have the notification signal set up, we'd probably want to use that for this assertion as it is slightly more robust (the status bar is a global object and might be modified by some other part of the GUI before we check, resulting in a false-positive test failure).